kafka/common_test.go

```diff
@@ -318,3 +324,59 @@ func TestTopicFieldFunc(t *testing.T) {
 		assert.Equal(t, zap.String("topic", "c"), topic)
 	})
 }
+
+// generateValidCACert creates a valid self-signed CA certificate in PEM format.
+func generateValidCACert(t testing.TB) []byte {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject:      pkix.Name{CommonName: "Test CA"},
+		NotBefore:    time.Now().Add(-time.Hour),
+		NotAfter:     time.Now().Add(time.Hour),
+		KeyUsage:     x509.KeyUsageCertSign | x509.KeyUsageKeyEncipherment,
+		IsCA:         true,
+	}
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &key.PublicKey, key)
+	require.NoError(t, err)
+
+	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: derBytes})
+}
+
+func TestTLSCACertPath(t *testing.T) {
```

Comment:
I think it would be good to add a test that verifies the CA is effective when dialling too. The code is simple enough that we could leave that till later though.